src/gui/invalidfilenamedialog.cpp

```diff
@@ -0,0 +1,169 @@
+
+    checkIfAllowedToRename();
+}
+
+InvalidFilenameDialog::~InvalidFilenameDialog() = default;
+
+void InvalidFilenameDialog::checkIfAllowedToRename()
+{
+    const auto propfindJob = new PropfindJob(_account, QDir::cleanPath(_folder->remotePath() + _originalFileName));
+    propfindJob->setProperties({ "http://owncloud.org/ns:permissions" });
+    connect(propfindJob, &PropfindJob::result, this, &InvalidFilenameDialog::onPropfindPermissionSuccess);
+    propfindJob->start();
+}
+
+void InvalidFilenameDialog::onPropfindPermissionSuccess(const QVariantMap &values)
+{
+    if (!values.contains("permissions")) {
+        return;
+    }
+    const auto remotePermissions = RemotePermissions::fromServerString(values["permissions"].toString());
+    if (!remotePermissions.hasPermission(remotePermissions.CanRename)
+        || !remotePermissions.hasPermission(remotePermissions.CanMove)) {
+        _ui->errorLabel->setText(
+            tr("You don't have the permission to rename this file. Please ask the author of the file to rename it."));
+        _ui->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(false);
+        _ui->filenameLineEdit->setEnabled(false);
+    }
+}
+
+void InvalidFilenameDialog::accept()
+{
+    _newFilename = _relativeFilePath + _ui->filenameLineEdit->text().trimmed();
+    const auto propfindJob = new PropfindJob(_account, QDir::cleanPath(_folder->remotePath() + _newFilename));
+    connect(propfindJob, &PropfindJob::result, this, &InvalidFilenameDialog::onPropfindSuccess);
+    connect(propfindJob, &PropfindJob::finishedWithError, this, &InvalidFilenameDialog::onPropfindError);
+    propfindJob->start();
+}
+
+void InvalidFilenameDialog::onFilenameLineEditTextChanged(const QString &text)
+{
+    const auto isNewFileNameDifferent = text != _originalFileName;
+    const auto illegalContainedCharacters = getIllegalCharsFromString(text);
+    const auto containsIllegalChars = !illegalContainedCharacters.empty() || text.endsWith(QLatin1Char('.'));
+    const auto isTextValid = isNewFileNameDifferent && !containsIllegalChars;
+
+    if (isTextValid) {
+        _ui->errorLabel->setText("");
+    } else {
+        _ui->errorLabel->setText(tr("Filename contains illegal characters: %1")
+                                     .arg(illegalCharacterListToString(illegalContainedCharacters)));
+    }
+
+    _ui->buttonBox->button(QDialogButtonBox::Ok)
+        ->setEnabled(isTextValid);
+}
+
+void InvalidFilenameDialog::onMoveJobFinished()
+{
+    const auto job = qobject_cast<MoveJob *>(sender());
+    const auto error = job->reply()->error();
+
+    if (error != QNetworkReply::NoError) {
+        _ui->errorLabel->setText(tr("Could not rename file. Please make sure you are connected to the server."));
+        return;
+    }
+
+    QDialog::accept();
+}
+
+void InvalidFilenameDialog::onPropfindSuccess(const QVariantMap &values)
+{
+    Q_UNUSED(values);
+
+    // File does exist. Can not rename.
```

Comment:
instead of adding this comment, should you not pick up a name that is self descriptive instead of `onPropFindSuccess` like `onAlreadyExists` ?